test/serialization.js

```diff
@@ -0,0 +1,57 @@
+'use strict';
```

Comment:
We should add explicit `source` to check against after serialization as a serialize\parse loop may introduce subtle bugs and not check all cases that may be somehow handled during parsing. (just to clarify, checks with serialize/parse loop should remain too)